app/assets/javascripts/views/vacation_request.js

```diff
@@ -0,0 +1,123 @@
+  el: 'section',
+  template: JST['templates/vacation_request'],
+
+  initialize: function() {
+    this.$el.html( this.template() );
+    // Prepare handy access for the controls
+    this.$available_days  = this.$('#available_days');
+    this.$vacation_type   = this.$('select[name=vacation_type]');
+
+    this.available_vacations  = new App.Collections.AvailableVacations();
+    this.vacation_requests    = new App.Collections.VacationRequests();
+    this.listenTo( this.available_vacations,  'reset',  this.render );
+    this.listenTo( this.vacation_requests,    'all',  this.logg );
+    // this.listenTo( this.collection, 'sync', this.render );
+    this.available_vacations.fetch({reset: true});
+    this.vacation_requests.fetch({reset: true});
+    // Set date picker variables
+    this.startDate  = new Date();
+    this.endDate    = this.startDate;
+    this.duration = 0;
+  },
+
+  render: function() {
+    // console.log(JSON.stringify(this.available_vacations));
+    // console.log(JSON.stringify(this.vacation_requests));
+    this.updateAvailableDays();
+    this.updateOkButton();
+
+    // TODO: Disable option as described below when there is no days available
+    // http://www.mkyong.com/jquery/how-to-set-a-dropdown-box-value-in-jquery/
+
+    return this;
+  },
+
+  events: {
+    'click  .ok':                         'onCreate',
+    'click  .cancel':                     'onCancel',
+    'change select[name=vacation_type]':  'onTypeChange',
+    'change input[name=start]':           'onStartDate',
+    'change input[name=end]':             'onEndDate',
+  },
+
+  calculateDuration: function( a, b ) {
+    d1 = Date.UTC(a.getFullYear(), a.getMonth(), a.getDate());
+    d2 = Date.UTC(b.getFullYear(), b.getMonth(), b.getDate());
+    var diff = d2 - d1;
+    // TODO: Swap values of controls if `startDate` is bigger than `endDate`
+    if (diff < 0) {
+      diff = -diff;
+      // this.$('input[name=start]').val(this.endDate.format);
+      // this.$('input[name=end]').val(this.startDate);
+    }
+    this.duration = (diff)/(1000*3600*24) + 1;
+    console.log( this.duration );
+    this.updateAvailableDays();
+  },
+
+  onStartDate: function( e ) {
+    this.startDate = new Date(e.target.value);
+    this.calculateDuration(this.startDate, this.endDate);
+  },
+
+  onEndDate: function( e ) {
+    this.endDate = new Date(e.target.value);
+    this.calculateDuration(this.startDate, this.endDate);
+  },
+
+  onTypeChange: function( e ) {
+    this.updateAvailableDays();
+    this.updateOkButton();
+  },
+
+  onCreate: function() {
+    console.log('onCreate');
+    this.vacation_requests.create({
+      'start':    this.startDate.getTime(),
+      'duration': this.duration,
+      'kind':     this.$vacation_type.val(),
+    });
+  },
+
+  onCancel: function( op ) {
+    console.log('onCancel' + op);
+  },
+
+  logg: function( e, p ) {
+    console.log(e);
+  },
+
+  updateOkButton: function() {
```

Comment:
It rather means 'disableOkButton', not sure why we need to disable button at all. 
What you do here is validator job, you can look at jquery.validator or something like this.